kratos/modeler/surrogate_boundary_modeler.h

```diff
@@ -0,0 +1,281 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ariadna Cortes Danes
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/node.h"
+#include "geometries/geometry.h"
+#include "voxel_mesh_generator_modeler.h"
+#include "spatial_containers/geometrical_objects_bins.h"
+#include "utilities/geometry_utilities/nearest_point_utilities.h"
+
+namespace Kratos
+{ 
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class SurrogateBoundaryModeler
+ * @ingroup KratosCore
+ * @brief Generates the voxel mesh plus stores the vector and signed distance from every node to the skin of the model.
+ */
+class SurrogateBoundaryModeler 
+    : public VoxelMeshGeneratorModeler
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef Node NodeType;
+    typedef Node::Pointer NodePtrType;
+    typedef Geometry<NodeType> GeometryType;
+    typedef GeometryType::Pointer GeometryPtrType;
+    typedef GeometryType::PointsArrayType PointsArrayType;
+
+    /// Pointer definition of SurrogateBoundaryModeler
+    KRATOS_CLASS_POINTER_DEFINITION( SurrogateBoundaryModeler );
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+    
+    ///Default constructor
+    SurrogateBoundaryModeler(){}
+
+    //Constructor
+    SurrogateBoundaryModeler(
+        Model& rModel, Parameters rParameters = Parameters()) 
```

Comment:
```suggestion
        Model& rModel,
        Parameters rParameters = Parameters()) 
```
Very minor stylish suggestion (more info [here](https://kratosmultiphysics.github.io/Kratos/pages/Kratos/For_Users/Conventions/Style_Guide.html)).